specutils/spectra/spectrum1d.py

```diff
@@ -194,6 +194,27 @@ def __truediv__(self, other):
         return self.divide(
             other, compare_wcs=lambda o1, o2: self._compare_wcs(self, other))
 
+    def __getitem__(self, key):
+        """
+        NDData is agnostic about self.spectral_axis so we need
+        to deal with it separately.
+
+        __setitem__ and __delitem__ are not duplicated in this way
+        as Spectrum1D does not support item assignment.
+        """
+
+        # Get a copy of the current spectral axis
+        spectral_axis = self.spectral_axis
+
+        # Call the sub-class (in particular, here call the NDData which
+        # does most fo the work.
+        self = super().__getitem__(key)
```

Comment:
Generally it's not a good idea to overwrite the `self` variable name - you might just use `other`.  And if you do that there's no need to "save" `spectral_axis` above - you can just do ``other.spectral_axis = self.spectral_axis[key]`` below.

Reply:
No problem on all, there are lots of subtleties here (at least to me).